modules/sdcore-control-plane-k8s/main.tf

```diff
@@ -0,0 +1,176 @@
+  source                 = "git::https://github.com/canonical/sdcore-pcf-k8s-operator//terraform"
+  model_name             = juju_model.sdcore.name
+  nrf_application_name   = module.sdcore-nrf-k8s.nrf_application_name
+  certs_application_name = module.self-signed-certificates.certs_application_name
+  db_application_name    = module.mongodb-k8s.db_application_name
+  channel                = var.channel
+}
+
+module "sdcore-smf-k8s" {
+  source                 = "git::https://github.com/canonical/sdcore-smf-k8s-operator//terraform"
+  model_name             = juju_model.sdcore.name
+  nrf_application_name   = module.sdcore-nrf-k8s.nrf_application_name
+  certs_application_name = module.self-signed-certificates.certs_application_name
+  db_application_name    = module.mongodb-k8s.db_application_name
+  channel                = var.channel
+}
+
+module "sdcore-udm-k8s" {
+  source                 = "git::https://github.com/canonical/sdcore-udm-k8s-operator//terraform"
+  model_name             = juju_model.sdcore.name
+  nrf_application_name   = module.sdcore-nrf-k8s.nrf_application_name
+  certs_application_name = module.self-signed-certificates.certs_application_name
+  channel                = var.channel
+}
+
+module "sdcore-udr-k8s" {
+  source                 = "git::https://github.com/canonical/sdcore-udr-k8s-operator//terraform"
+  model_name             = juju_model.sdcore.name
+  nrf_application_name   = module.sdcore-nrf-k8s.nrf_application_name
+  certs_application_name = module.self-signed-certificates.certs_application_name
+  db_application_name    = module.mongodb-k8s.db_application_name
+  channel                = var.channel
+}
+
+module "sdcore-webui-k8s" {
+  source              = "git::https://github.com/canonical/sdcore-webui-k8s-operator//terraform"
+  model_name          = juju_model.sdcore.name
+  db_application_name = module.mongodb-k8s.db_application_name
+  channel             = var.channel
+}
+
+module "mongodb-k8s" {
+  source       = "git::https://github.com/canonical/mongodb-operator.git//terraform"
+  model_name   = juju_model.sdcore.name
+  channel      = var.mongo-channel
+  mongo-config = var.mongo-config
+}
+
+module "grafana-agent-k8s" {
+  source                         = "git::https://github.com/canonical/grafana-agent-k8s-operator//terraform"
+  model_name                     = juju_model.sdcore.name
+  channel                        = var.grafana-channel
+  metrics_remote_write_offer_url = var.metrics_remote_write_offer_url
+  logging_offer_url              = var.logging_offer_url
+  grafana-config                 = var.grafana-config
+}
+
+module "self-signed-certificates" {
+  source      = "git::https://github.com/canonical/self-signed-certificates-operator.git//terraform"
+  model_name  = juju_model.sdcore.name
+  channel     = var.cert-channel
+  cert-config = var.cert-config
+}
+
+module "traefik-k8s" {
+  source         = "git::https://github.com/canonical/traefik-k8s-operator//terraform"
```

Comment:
Source?

Reply:
It's same for all the charms. The source is terraform subdirectory in the main branch of charm.